packages/discovery-provider/src/queries/query_helpers.py

```diff
@@ -616,6 +750,8 @@ def getContentId(metadata):
         session, gated_content_access_args
     )
 
+    is_authorized_as_user = is_authorized_request(current_user_id)
```

Comment:
Just a thought here: I'm generally opposed to putting code in `queries` that reads or uses the request context if we can help it. This has led to some nastiness in the past (reading current user id from the `X-UserId` header for example). Keeping the clean separation of responsibilities has made stuff more testable and reusable. 

I'd prefer we keep the request context parsing logic up at the API level and restrict the `queries` folder's knowledge to just interactions with the database given a set of parameters.

I recognize that it's a bit of a pain to move it up to that level and it can certainly happen later if we're in agreement.